packages/suite/src/components/suite/ChangeDeviceLabel.tsx

```diff
@@ -0,0 +1,123 @@
+import { useDevice, useDispatch, useTranslation } from 'src/hooks/suite';
+import * as deviceSettingsActions from 'src/actions/settings/deviceSettingsActions';
+import { MAX_LABEL_LENGTH } from 'src/constants/suite/device';
+import { SCREEN_SIZE } from '@trezor/components/src/config/variables';
+import { isAscii } from '@trezor/utils';
+
+const StyledInput = styled(Input)<{ isVertical?: boolean }>`
+    ${props =>
+        props.isVertical &&
+        css`
+            margin: 4px 0 4px 4px;
+
+            &:not(:first-child) {
+                margin-left: 8px;
+            }
+
+            @media (max-width: ${SCREEN_SIZE.SM}) {
+                margin: 0px 0 10px;
+                width: 100%;
+            }
+        `}
+`;
+
+const StyledButton = styled(Button)<{ isVertical?: boolean; isDisabled: boolean }>`
+    height: ${INPUT_HEIGHTS.large}px;
+
+    ${props =>
+        props.isVertical &&
+        css`
+            min-width: 170px;
+            margin: 4px 0 4px 4px;
+            cursor: ${props.isDisabled ? 'not-allowed' : 'pointer'};
+            &:not(:first-child) {
+                margin-left: 8px;
+
+                @media (max-width: ${SCREEN_SIZE.SM}) {
+                    margin-left: 0;
+                }
+            }
+
+            @media (max-width: ${SCREEN_SIZE.SM}) {
+                width: 100%;
+                margin: 0;
+            }
+        `}
+`;
+
+interface ChangeDeviceLabelProps {
+    isDeviceLocked: boolean;
+    placeholder?: string;
+    isVertical?: boolean;
+    onClick?: () => void;
+}
+
+export const ChangeDeviceLabel = ({
+    isDeviceLocked,
+    placeholder,
+    isVertical,
+    onClick,
+}: ChangeDeviceLabelProps) => {
+    const { translationString } = useTranslation();
+    const { device } = useDevice();
+    const dispatch = useDispatch();
+
+    const [label, setLabel] = useState(device?.label === placeholder ? '' : device?.label);
+    const [error, setError] = useState<string | null>(null);
+
+    const handleChange: React.ChangeEventHandler<HTMLInputElement> = ({ target: { value } }) => {
+        setLabel(value);
+
+        if (value.length > MAX_LABEL_LENGTH) {
+            setError(
+                translationString('TR_LABEL_ERROR_LENGTH', {
+                    length: 16,
+                }),
+            );
+        } else if (!isAscii(value)) {
+            setError(translationString('TR_LABEL_ERROR_CHARACTERS'));
+        } else {
+            setError(null);
+        }
+    };
+
+    const handleClick = () => {
+        dispatch(deviceSettingsActions.applySettings({ label }));
+        analytics.report({
+            type: EventType.SettingsDeviceChangeLabel,
+        });
+        onClick?.();
+    };
+
+    return (
+        <>
+            <StyledInput
+                width={170}
+                isVertical={isVertical}
+                noTopLabel
+                bottomText={error}
+                value={label}
+                placeholder={placeholder || ''}
```

Comment:
No problem if it is `undefined`.

```suggestion
                placeholder={placeholder}
```